src/utils/types/interceptors.ts

```diff
@@ -1,17 +1,19 @@
+import type { Request, Response } from 'express';
+
 export interface InterceptorRequestParams {
-  request: import('express').Request;
+  request: Request;
 }
 
 export type InterceptorRequest = (params: InterceptorRequestParams) => void;
 
 export interface InterceptorResponseParams {
-  request: import('express').Request;
-  response: import('express').Response;
+  request: Request;
+  response: Response;
   setDelay: (delay: number) => Promise<void>;
   setStatusCode: (statusCode: number) => void;
 }
 
-export type InterceptorResponse = <Data>(data: Data, params: InterceptorResponseParams) => any;
+export type InterceptorResponse<DataType extends any = any> = (data: DataType, params: InterceptorResponseParams) => any;
```

Comment:
Походу ты забыл удалить, тут дженерик не нужен же

Reply:
наоборот нужен, чтобы можно было свой кинуть при объявлении интерцепотора и с any не работать

Reply:
При объявлении невозможно прокинуть generic. Об этом же и говорилось в треде stackoverflow который я скидывал. А any переопределяется с помощью `arg: ${type instead of any}` - это будет работать, т.к. можно ставить типы, которые входят в изначальный, а в any входит все

Reply:
дженерик можно использовать вне конфига, например у тебя функция в другом файле

Reply:
<img width="1057" alt="image" src="https://user-images.githubusercontent.com/47690223/233792276-ebb082b2-74b0-4c9a-8451-3e61c4f3ae83.png">
вот спокойно получилось, по сути для такого кейса дженерик и остался